bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ExpressionThreadPoolManager.java

```diff
@@ -225,8 +225,9 @@ public void schedule(final Runnable task, final Expression expression) {
 
         @Override
         public boolean remove(Runnable task) {
-            if (futures.get(task) != null && futures.get(task).cancel(false)) {
-                running.remove(task);
+            Future<?> future = futures.get(task);
+            if (future != null && future.cancel(false)) {
+                running.remove(future);
```

Comment:
I think it should be:
        
@Override
        public boolean remove(Runnable task) {
            Future<?> future = futures.get(task);
            if (future != null && future.cancel(false)) {
                futures.remove(future);
            }
            running.remove(task);
            return super.remove(task);
        }

running holds Runnables.